### PR TITLE
68 Create Wallets' boundaries and tests

### LINKED ISSUE #68: Create Wallets' boundaries and tests
ℹ️ This issue is part of the Epic #66 

### Describe the solution you'd like

Implement the Wallets' boundaries and tests.

### Describe alternatives you've considered
- Create a module in the path `apps/mintacoin/lib/mintacoin/wallets/wallets.ex` to define exposed functions.
- Create the tests for Wallets' boundaries

### PR DESCRIPTION
**Pull request by MrJuanGaviriaK**
Closes #68 

## Description

Implement the Wallets' boundaries and tests.

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/mintacoin/test/mintacoin/wallets/wallets_test.exs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  @moduledoc """
+    This module is used to group common tests for Wallets functions
+  """
+
+  use Mintacoin.DataCase, async: false
+
+  import Mintacoin.Factory
+
+  alias Mintacoin.{Repo, Wallet, Wallets, Account, Accounts, Blockchain, Blockchains}
+  alias Blockchains.Network
```
Comment (miguelnietoa) [inline apps/mintacoin/test/mintacoin/wallets/wallets_test.exs:11]:
```suggestion
  alias Mintacoin.{Repo, Wallet, Wallets, Account, Accounts, Blockchain, Blockchains, Blockchains.Network}
```

**Review by miguelnietoa — CHANGES_REQUESTED** [review]
Very good job @MrJuanGaviriaK 💯 

Please, address the requested changes and that's it!

File: apps/mintacoin/lib/mintacoin/wallets/wallets.ex [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  @spec retrieve(id :: id()) :: {:ok, Wallet.t()} | {:error, error()}
+  def retrieve(id), do: retrieve_by(id: id)
+
+  @spec retrieve_by_address(address :: address()) :: {:ok, Wallet.t()} | {:error, error()}
+  def retrieve_by_address(address), do: retrieve_by(address: address)
+
+  @spec retrieve_by_account_id(account_id :: account_id()) ::
+          {:ok, Wallet.t()} | {:error, error()}
+  def retrieve_by_account_id(account_id), do: retrieve_by(account_id: account_id)
```
[the remark this answers is not in the document]
Comment (MrJuanGaviriaK) [reply apps/mintacoin/lib/mintacoin/wallets/wallets.ex:31]: I'm going to fix it! Thanks.

File: apps/mintacoin/lib/mintacoin/wallets/wallets.ex [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  @moduledoc """
+  This module is responsible for doing the CRUD operations for Wallets
+  """
+  alias Ecto.{Query.CastError, NoResultsError, Changeset, UUID}
+  alias Mintacoin.{Repo, Wallet}
+
+  @type id :: UUID.t()
+  @type address :: String.t()
+  @type account_id :: String.t()
+  @type blockchain_id :: String.t()
```
Comment (miguelnietoa) [inline apps/mintacoin/lib/mintacoin/wallets/wallets.ex:11]:
```suggestion
  @type account_id :: UUID.t()
  @type blockchain_id :: UUID.t()
```

**Review by miguelnietoa — CHANGES_REQUESTED** [review]

**Review by miguelnietoa — APPROVED** [review]

**Pull request merged**